src/polynomial.h

```diff
@@ -0,0 +1,69 @@
+#ifndef SYMENGINE_POLYNOMIALS_H
+#define SYMENGINE_POLYNOMIALS_H
+
+#include "monomials.h"
+#include "dict.h"
+#include "basic.h"
+#include "integer.h"
+
+namespace SymEngine {
+
+    class Polynomial : public Basic{
+    public:
+        int degree;
+        std::string var_;
+        map_uint_mpz dict_;
+    public:
+        IMPLEMENT_TYPEID(POLYNOMIAL)
+        Polynomial(){}
+
+        Polynomial(const std::string& var, map_uint_mpz&& dict);
+
+        std::size_t __hash__() const {return 0;}
+        //{
+            //std::hash<long long int> hash_fn;
+            //return hash_fn(this->var_);
+        //}
+
+        bool __eq__(const Basic &o) const
+        {
+            if ((var_ == static_cast<const Polynomial &>(o).var_) &&
+                map_uint_mpz_eq(dict_, static_cast<const Polynomial &>(o).dict_))
+                return true;
+
+            return false;
```

Comment:
Move these methods to .cpp. The lest code is in headers, the faster compilation and less clutter.